pkg/apis/core/validation/validation.go

```diff
@@ -4053,6 +4053,35 @@ func ValidateService(service *core.Service) field.ErrorList {
 		ports[key] = true
 	}
 
+	// Validate TopologyKeys
+	if len(service.Spec.TopologyKeys) > 0 {
+		topoKeys := sets.NewString()
+		topoPath := specPath.Child("topologyKeys")
+		// topologyKeys is mutually exclusive with 'externalTrafficPolicy=Local'
+		if service.Spec.ExternalTrafficPolicy == core.ServiceExternalTrafficPolicyTypeLocal {
+			allErrs = append(allErrs, field.Forbidden(topoPath, "may not be specified when `externalTrafficPolicy` is 'Local'"))
```

Comment:
Yes, gonna mention that, and may have more discussion in later PRs.

Reply:
Caveat: externalTrafficPolicy triggers healthcheck changes in kube-proxy, topology does not.  It may be that we should make those more universal, but that's the way it is now.

Reply:
pkg/registry/core/service/storage/rest.go `func allocateHealthCheckNodePort`